internal/services/azurestackhci/stack_hci_storage_path_resource.go

```diff
@@ -0,0 +1,238 @@
+	return "azurerm_stack_hci_storage_path"
+}
+
+func (StackHCIStoragePathResource) ModelObject() interface{} {
+	return &StackHCIStoragePathResourceModel{}
+}
+
+type StackHCIStoragePathResourceModel struct {
+	Name              string                 `tfschema:"name"`
+	ResourceGroupName string                 `tfschema:"resource_group_name"`
+	Location          string                 `tfschema:"location"`
+	CustomLocationId  string                 `tfschema:"custom_location_id"`
+	Path              string                 `tfschema:"path"`
+	Tags              map[string]interface{} `tfschema:"tags"`
+}
+
+func (StackHCIStoragePathResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"custom_location_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: customlocations.ValidateCustomLocationID,
+		},
+
+		"path": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (StackHCIStoragePathResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r StackHCIStoragePathResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.StorageContainers
+
+			var config StackHCIStoragePathResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := storagecontainers.NewStorageContainerID(subscriptionId, config.ResourceGroupName, config.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := storagecontainers.StorageContainers{
+				Name:     pointer.To(config.Name),
+				Location: location.Normalize(config.Location),
+				Tags:     tags.Expand(config.Tags),
+				ExtendedLocation: &storagecontainers.ExtendedLocation{
+					Name: pointer.To(config.CustomLocationId),
+					Type: pointer.To(storagecontainers.ExtendedLocationTypesCustomLocation),
+				},
+				Properties: &storagecontainers.StorageContainerProperties{
+					Path: config.Path,
+				},
+			}
+
+			future, err := client.CreateOrUpdate(ctx, id, payload)
```

Comment:
why not use `CreateOrUpdateThenPoll` ?

Reply:
the resource might exist if poll fail, so here we setID before poll.

Reply:
why not put `metadata.SetID(id)` before or after `CreateOrUpdateThenPoll` ?

Reply:
The resource creation might fail before the instance is created (e.g., a same name resource already exists, or name property invalid), in this case the ID should not be saved.

Reply:
both `creation failure` and `poll failure` are failure, why `SetID` for one but not for the other? We should set ID after `CreateOrUpdateThenPoll` to keep consistent with other resources

Reply:
for this resource, `creation failure` will not create the instance, but  `poll failure` will result in the resource created but with a failed status, if we save the ID for `poll failure`, it will save effort for users to manually import or delete the failed instance. This is similar to the case in https://github.com/hashicorp/terraform-provider-azurerm/pull/25646#discussion_r1593575660.